### PR TITLE
feat: add injected wallets to the UX

### PR DESCRIPTION
**Pull request by moldy530**
# Pull Request Checklist

This adds a button group class and fixes up the injected wallets config so that we can render them in the UX

- [ ] Did you add new tests and confirm existing tests pass? (`yarn test`)
- [ ] Did you update relevant docs? (docs are found in the `site` folder, and guidelines for updating/adding docs can be found in the [contribution guide](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md))
- [ ] Do your commits follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] Does your PR title also follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] If you have a breaking change, is it [correctly reflected in your commit message](https://www.conventionalcommits.org/en/v1.0.0/#examples)? (e.g. `feat!: breaking change`)
- [ ] Did you run lint (`yarn lint:check`) and fix any issues? (`yarn lint:write`)
- [ ] Did you follow the [contribution guidelines](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md)?

---

## PR-Codex overview
The focus of this PR is to enhance authentication configuration in the UI by adding support for passkey authentication and external wallets display options.

### Detailed summary
- Added support for passkey authentication in UI configuration
- Enhanced UI to display external wallets based on configuration
- Updated button styling in UI components
- Improved authentication methods in UI components

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by moldy530** [comment]
image
image

File: account-kit/react/src/components/auth/card/main.tsx
Comment (rthomare) [inline account-kit/react/src/components/auth/card/main.tsx:17]: lol this is slightly conufsing. So if sign in text is true or null we'd want to show signin text? Is there a better default we could use here?

File: account-kit/react/src/components/auth/sections/EmailAuth.tsx
Comment (rthomare) [inline account-kit/react/src/components/auth/sections/EmailAuth.tsx:8]: where is this used?

File: account-kit/react/src/components/auth/sections/InjectedProvidersAuth.tsx
Comment (rthomare) [inline account-kit/react/src/components/auth/sections/InjectedProvidersAuth.tsx:7]: Does this not apply anymore?

File: account-kit/react/src/components/button.tsx
Comment (rthomare) [inline account-kit/react/src/components/button.tsx:51]: Sorry a little confused what is the thing we are trying to achieve with the UI. To me this feels a little in-direct. If a child view is reaching out to understand parent context that feels like a code smell. Especially in a broad component like button.

File: account-kit/react/src/components/auth/sections/InjectedProvidersAuth.tsx
Comment (moldy530) [reply account-kit/react/src/components/auth/sections/InjectedProvidersAuth.tsx:7]: now that all the exported methods are correctly being parsed, we shouldn't need to disable this lint rule anymore

File: account-kit/react/src/components/button.tsx
Comment (moldy530) [reply account-kit/react/src/components/button.tsx:51]: I agree generally about not relying on your parents for styling your self, but here this was probably the easiest way to achieve this layout:
image

There's a couple wrinkles:
1. since a button group is max 2 rows, I did this by having a flex wrap + setting the first child at max width and the rest of them having a flex grow and flex shrink. This is done on the `btn-group` class. Because the children are flex grow and shrink and the container is flex wrap, the buttons will shrink only until the text is not cut off. So we can't do something in the button like `if (width < text.width + icon.width) hideText` because that will never be the case and the buttons will overflow into the next row.
2. If we made a button group component, the component itself wouldn't be able to hold this logic because it would just receive children and AFAIK it's not straightforward to check what kind of child it is and pass in new props
3. the button group is just the high level container for each auth section. Almost all of the sections are buttons so that's chill, but you can also have email as a section. The email section is typically rendered in the first section and someone could add other sections along side email. In that case, you would want the remaining sections (which are all buttons) to render as the second row of a button group.

This ultimately felt like the easiest thing to do to match the layout. Buttons aren't actually exported to be consumed externally, so this is all internal to the SDK which feels like an ok place to make these kinds of one offs

File: account-kit/react/src/components/auth/sections/EmailAuth.tsx
Comment (moldy530) [reply account-kit/react/src/components/auth/sections/EmailAuth.tsx:8]: In the component itself, this moved because I ran the linter which re-ordered imports

File: account-kit/react/src/components/button.tsx
Comment (dphilipson) [reply account-kit/react/src/components/button.tsx:51]: Hm, this is pretty tricky. So if I'm understanding correctly, the goal of this code is to apply different styling if the button has more than three siblings (counting itself)?

I'm okay with this solution, but did you look into using tricky CSS selectors to apply based on sibling count? I think it might be possible to do something like
```css
.parent:has(> :nth-child(4)) > * {
    /* ... */
}
```
where that rule would match elements that are a child of a `.parent` element if that `.parent` has at least four children. If that works, it might be a little simpler.

File: account-kit/react/src/tailwind/components/buttons.ts
Comment (dphilipson) [inline account-kit/react/src/tailwind/components/buttons.ts:4]: For my own understanding, how are these CSS rules localized to the account-kit UI component? I'm assuming this isn't changing global stylings for common classes like `.btn`, is it?

File: account-kit/react/src/components/button.tsx
Comment (moldy530) [reply account-kit/react/src/components/button.tsx:51]: oh that's cool, lemme try that

File: account-kit/react/src/tailwind/components/buttons.ts
Comment (moldy530) [reply account-kit/react/src/tailwind/components/buttons.ts:4]: I need to double check, I think they might conflict with global styles actually. I need to double check, but I think we might need to configure our plugin to use a prefix to avoid the clashing

File: account-kit/react/src/components/button.tsx
Comment (moldy530) [reply account-kit/react/src/components/button.tsx]: ok sweet! check it out now? I moved it into css, though I think the smell still exists here because it does require specific set up between btn and btn groups, but I think this is ok. At somepoint, we will need to support the case where there are more than 4 buttons in the second row. What we will have to do there is: if there are 4 buttons, they get condensed into a `...` button that opens some other view containing the overflow. To handle that we'll have to use a `ref` that checks either sibling counts or children, so we might not be out of the woods fully. That said I like the css solution more because it keeps all of this btn and btn group styling in the same place, so it's a bit more discoverable

File: account-kit/react/src/tailwind/components/buttons.ts
Comment (moldy530) [reply account-kit/react/src/tailwind/components/buttons.ts:4]: this probably is a problem, but will have to deal with it in a separate PR

File: account-kit/react/src/components/button.tsx
Comment (dphilipson) [reply account-kit/react/src/components/button.tsx]: That sort of thing may still be possible with CSS if the "..." button is always in the DOM but hidden by CSS unless there are more than four buttons in the second row. But we can cross that bridge when we come to it. The new version looks good!

**Review by dphilipson — DISMISSED** [review]
Looks good with the latest change, but presumably needs some kind of rebase. The changeset is so large that GitHub’s UI is lagging as I type this.

**Comment by rthomare** [comment]
Also LGTM but let me know if you need a review after the rebase

**Pull request merged**
